[Docs] Getting started guide

laggron42 opened the pull request:
### Type

- [ ] Bugfix
- [x] Enhancement
- [ ] New feature

### Description of the changes

As said in #1757, I'm splitting the user documentation I'm working on in multiple pull requests. This guide is an updated version of [the V2 getting started doc](http://twentysix26.github.io/Red-Docs/red_getting_started/), including a quick tutorial for hosting on a VPS (less "how to make bot always connected" questions in #support).

laggron42 committed:
Getting started guide

mikeshardmind commented on docs/host-list.rst:
```diff
@@ -0,0 +1,81 @@
+|<https://www.scaleway.com/>`_       |`<https://online.net/>`_, based in Europe.            |
++------------------------------------+------------------------------------------------------+
+|`DigitalOcean                       |US-based cheap VPSes. The gold standard. Locations    |
+|<https://m.do.co/c/71288f6af536/>`_ |available world wide.                                 |
```
I don't mind if we want to offer a referral link, but we should be explicit that it is one if we are going to do this.

Tobotimus replied:
Whose referral link is this? @retke's? The recipient of the benefits going to the link owner should only ever be the Cog Creators organisation... I understand he's putting a lot of work into these docs but putting a link in there that makes the contributor money is unfair and against the spirit of open source development...

mikeshardmind replied:
For the record, this is the DO referral link we have as a CC on the discord server. ( I went to compare it when I saw, and my response would have been different had it not been)

Tobotimus replied:
Right. Do we know who it belongs to?

mikeshardmind replied:
Pull discussion back to discord on this, but for now, the referral link should be removed.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+    [p]help
+
+.. tip:: The message is generated dynamically and users will only see the
+    commands they can use. You can edit this using the permissions cog.
```
"You can edit this using the permissions cog.", what "this" refers to is a bit vague. Maybe reword this to "You can change what commands users can use with the permissions cog."

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+.. tip:: The message is generated dynamically and users will only see the
+    commands they can use. You can edit this using the permissions cog.
+
+You can also pick a command to get its detailled description and the
```
"detailled" is misspelled, should be "detailed"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+
+.. code-block:: none
+
+    [p]help command
```
This should be changed to `[p]help <command>` to show that <command> should be replaced.
I don't want to add a comment every place you have this, so make sure you look for all of them.

laggron42 replied:
There's a difference between the command/subcommand name and their arguments. This tutorial should be very explicit as it is the first and most important one, and we don't want users to type `[p]help <play>`, it is explained right below how command arguments are documented. Also this is very similar to the [V2 Getting Started guide](http://twentysix26.github.io/Red-Docs/red_getting_started/), trying to do the same for this part.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+The parameters are shown as enclosed in ``< >`` if they're needed, or
+``[ ]`` if optional.
+As an example, the ban command will show this in the help message:
+``Syntax: !ban <user> [days] [reason]``
```
Not sure why you used `!` as the prefix instead of `[p]` here.

laggron42 replied:
As said above, it's an example, and `!` is the most common prefix.

Flame442 replied:
This looks like the only place (besides where you explain what `[p]` means) that you use `!` as a prefix instead of `[p]`. Although it does say that it is an example, it also says "the ban command __will__ show this in the help message" which at least to me seems to imply it will always say that. You should at least change the line above to say something like "As an example, if your prefix is set to '!', the ban command will show this in the help message:".

Flame442 replied:
This still is not resolved.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+
+    [p]help commandgroup subcommand
+
+When using them, you also need to specify the command group.
```
Again, what "them" is referring to isn't very clear. This should be reworded to something like "When using commands that have subcommands,"

laggron42 replied:
Wouldn't work with "need to specify the command group". Modified it a bit tho.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+----
+
+Red is built with cogs, fancy term for plugins. They are
+modules that enhance the Red functionnalities. They contain
```
"functionnalities" is misspelled, should be "functionalities"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+
+        [p]load cog1 cog2 ...
+
+You can enable and disable everything you want, which means you can
```
Instead of saying enable and disable, this should probably say load and unload so that it uses the terminology Red uses.

laggron42 replied:
This isn't supposed to show how to do it but to be clear about what it adds to red : something customizable where you can enable and disable whatever you want, "load" and "unload" wouldn't be clear in this kind of sentences.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+~~~~~~~~~~~~~~
+
+There's an entire `community <https://discord.gg/red>`_ that contributes
+to Red. Those contibutors make additional cogs for you to use. You can
```
"contibutors" is misspelled, should be "contributors"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+You can find cogs by searching on `cogs.red <https://cogs.red>`_. Find whatever you want,
+there are hundreds of cogs available!
+
+.. note:: ``cogs.red``, the website that list all of the cogs is not
```
I'm not sure if this should even include cogs.red until that is up, no reason to confuse the users tho don't bother reading the note or do not know what bot version they have installed.

laggron42 replied:
I was thinking about removing the note as soon as cogs.red is ready, so users would know a better solution is in progress. I can remove the hyperlink to prevent confusion tho.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+exclusive commands that can interact with the global settings
+or system files.
+
+If it *you* by default. If not set, use ``[p]set owner``
```
"If it *you* by default." is a fragment and "If not set, use" just sounds bad.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+~~~~~~~~~~~~~
+
+The administrator is defined by its roles. You can set an admin role
+with the ``[p]set adminrole`` command.
```
This command was changed with #2783.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+~~~~~~~~~
+
+A moderator is a step above the average users. You can set a moderator
+role with the ``[p]set modrole`` command.
```
This command was changed with #2783.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+Hosting
+-------
+
+If you are hosting Red on your personnal computer, you will soon notice that
```
"personnal" is misspelled, should be "personal"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+
+If you are hosting Red on your personnal computer, you will soon notice that
+if you close the window or if you shut down you computer, Red will be offline.
+He needs an environement to work and respond.
```
I think Red is considered a "she".
"environement" is misspelled, should be "environment"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+He needs an environement to work and respond.
+
+You can try to host Red somewhere he will always be online, like on a virtual
+private server (VPS) or on a personnal server (e.g. Raspberry Pi).
```
"personnal" is misspelled, should be "personal"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+If you want to do it, follow these steps.
+
+.. warning::
+    Before trying to host Red on a Linux environement, you need to know the
```
"environement" is misspelled, should be "environment"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+
+.. warning::
+    Before trying to host Red on a Linux environement, you need to know the
+    basics of the Unix commands, such as navigate the system files or use
```
"such as navigate" is not grammatically correct.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+
+    You should follow `this guide
+    <https://www.digitalocean.com/community/tutorials/an-introduction-to-linux-basics>`_
+    from DigitalOcean which will introduct you to the Linux basics.
```
Should be "introduce" not "introduct".

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+  quality VPS providers.
+
+  You can also buy a Raspberry Pi (~$20), which is a micro-computer that will
+  be able to host Red. The model 3 or above is recommanded.
```
"recommanded" is misspelled, should be "recommended"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+    if want to restrict a cog or a command to a channel/member, you can use
+    the permissions cog.
+
+.. _getting-started-hosting:
```
Not sure why this section is here, this guide is for after the user installs the bot, so why should they be told how to host the bot *now*?

laggron42 replied:
Because the first question is usually "why do my bot disconnects when I shutdown my computer", and this ends up in a painful "how to setup VPS" tuto in #support

Flame442 replied:
I don't think this should be here, but I also don't know where this could go that would be better. If we has a FAQ this would be perfect for that, but I don't believe anything like that exists.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+2. **Install Linux**
+
+  Most of the VPS providers have tools for installing Linux automatically. If
+  you're a beginner, we recommand **Ubuntu 18**.
```
"recommand" is misspelled, should be "recommend"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+User documentation
+------------------
+
+You will soon start using the Red core cogs. A detailled documentation is
```
"detailled" is misspelled, should be "detailed"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+
+* **Guide**
+
+  This will introduct you to the cog and explain you how it works.
```
Should be "introduce" not "introduct".

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+
+  * **Description**
+
+    A detailled description of what the command does, with details about how
```
"detailled" is misspelled, should be "detailed"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,361 @@
+        of days of messages to be deleted will be equal to ``1`` if not
+        specified.
+
+* **Frequently asked questions**
```
Why add this if nothing is there?

laggron42 replied:
Was supposed to be used at first, forgot to remove it.

Flame442 commented on docs/host-list.rst:
```diff
@@ -0,0 +1,81 @@
+    of the :ref:`getting started <getting-started>` guide. Please take a look
+    if you don't know how to host Red on a VPS.
+
+This is a list of the recommanded VPS providers.
```
"recommanded" is misspelled, should be "recommended"

Flame442 commented on docs/host-list.rst:
```diff
@@ -0,0 +1,81 @@
+This is a list of the recommanded VPS providers.
+
+.. warning::
+    Please be aware that a Linux server is controlled through command line.
```
"through command line." is not grammatically correct. "through a command line." would be better.

Flame442 commented on docs/host-list.rst:
```diff
@@ -0,0 +1,81 @@
+    Please be aware that a Linux server is controlled through command line.
+    If you don't know Unix basics, please take a look at `this guide
+    <https://www.digitalocean.com/community/tutorials/an-introduction-to-linux-basics>`_
+    from DigitalOcean which will introduct you to the Linux basics.
```
Should be "introduce" not "introduct".

Flame442 commented on docs/host-list.rst:
```diff
@@ -0,0 +1,81 @@
+|`DigitalOcean                       |US-based cheap VPSes. The gold standard. Locations    |
+|<https://m.do.co/c/71288f6af536/>`_ |available world wide.                                 |
++------------------------------------+------------------------------------------------------+
+|`OVH <https://www.ovh.co.uk/>`_     |Cheap VPSes, used by many people. France and Canadian |
```
Should probably be "French and Canadian" to keep the tense (?) the same.

laggron42 committed:
Remove DigitalOcean referral link

laggron42 committed:
Fix typos and mispells, thanks to @Flame442

laggron42 committed:
Remove cogs.red hyperlink until it is finished

Tobotimus committed:
Merge branch 'V3/develop' into docs/getting_started

laggron42 committed:
Add towncrier entry

laggron42 committed:
Add prolog.txt

This is not necessary for this PR but all of the other cog guides rely on this file.
The cog guides are individual branches based on this one, which is why I'm adding this file right now.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+As a chatbot, you interact with Red via the Discord text channels
+(not from the command prompt). For that, you will use the prefix you
+set before. For example, if your prefix is ``!``, you will execute your
+command like this: ``!ping``.
```
The sentence starting with "For that" should probably be reworded to something like "To send commands to the bot, use the prefix you set before followed by the command you want to use.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+
+    [p]help command
+
+The parameters are shown as enclosed in ``< >`` if they're needed, or
```
"needed" should be changed to "required"

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+``Syntax: !ban <user> [days] [reason]``
+
+This means that it is necessary to provide ``user``. However, the
+``days`` value (number of messages to delete) is optional, like
```
Saying "like the reason value" implies that it was previously know that the reason value was optional. Reword to something like "as well as the reason value".

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+``days`` value (number of messages to delete) is optional, like
+the ``reason`` value, used for the modlog.
+
+You can use help to show the **categories** too (named cogs).
```
"named cogs" is a tad confusing, maybe change it to "also called cogs". The parentheses could also be removed.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+the ``reason`` value, used for the modlog.
+
+You can use help to show the **categories** too (named cogs).
+Just do so (notice the capitalization):
```
"Just do so" is a bit of a sentence fragment. This needs to be reworded to sound better.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+exclusive commands that can interact with the global settings
+or system files.
+
+*You* are the owner by default. If the owner isn't set, use ``[p]set owner``.
```
`[p]set owner` is removed as of #2928.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+if you close the window or if you shut down you computer, Red will be offline.
+She needs an environment to work and respond.
+
+You can try to host Red somewhere he will always be online, like on a virtual
```
I believe Red is considered a "she", but it would be best to just not use gendered pronouns.

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+    A list of all arguments needed (or not) for the command, with more details.
+
+    .. tip::
+        Arguments enclosed in ``< >`` means that the argument is **necessary**
```
Like above, this should probably be "required".

Flame442 commented on docs/getting_started.rst:
```diff
@@ -0,0 +1,358 @@
+        Arguments enclosed in ``< >`` means that the argument is **necessary**
+        for the command to work.
+
+        Arguments enclosed in ``[ ]`` means that the command is **optional**
```
The **argument** is optional.

laggron42 committed:
QA changes

Flame442 commented on docs/getting_started.rst:
```diff
@@ -80,17 +81,18 @@ parameters.
+The parameters are shown as enclosed in ``< >`` if they're required, or
 ``[ ]`` if optional.
-As an example, the ban command will show this in the help message:
+As an example, the ban command will show this in the help message, considering
```
"Assuming" would be better than "considering".

Flame442 commented on docs/getting_started.rst:
```diff
@@ -240,7 +242,7 @@ Hosting
-She needs an environment to work and respond.
+It needs an environment to work and respond.
 
 You can try to host Red somewhere he will always be online, like on a virtual
```
You changed a "she" to "it", but didn't change the "he". After the discussion in AC, both of these should probably be she.

laggron42 committed:
More QA changes

considering -> assuming
red is a girl, not non-binary
